File: problems/This_is_coding_test/1_X/3_2_큰 수의 법칙.py
N, M, K = map(int, input().split())
nums = list(map(int, input().split()))

nums.sort(reverse=True)
largest1 = nums[0]
largest2 = nums[1]

rep_count = int(M / (K + 1))
rep_num = largest1 * K + largest2

remaining_count = M % (K + 1)

answer = rep_num * rep_count + largest1 * remaining_count
print(answer)

File: problems/This_is_coding_test/2_X/3_2 큰 수의 법칙.js
/**
 * @param {string[]} inputs
 */
function solution(inputs) {
  const [N, M, K] = inputs[0].split(" ").map((value) => parseInt(value));
  let numArr = inputs[1].split(" ").map((value) => parseInt(value));

  numArr.sort();
  const largest1 = numArr[N - 1];
  const largest2 = numArr[N - 2];

  const repCount = Math.floor(M / (K + 1));
  const remains = M % (K + 1);

  let sum = repCount * (largest1 * K + largest2);
  sum += remains * largest1;
  console.log(sum);
}

const rl = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});
let inputs = [];
rl.on("line", (line) => {
  inputs.push(line);
}).on("close", () => {
  solution(inputs);
  process.exit();
});

// 5 8 3
// 2 4 5 4 6
// # 46

// 5 7 2
// 3 4 3 4 3
// # 28
